Dockerfile

```
@@ -0,0 +1,31 @@
+# select build image
+FROM rust as builder
```

This is neat, I hadn't seen multi-stage docker builds before (granted I haven't really used docker for maybe ~6 years).

Based on [this link](https://docs.docker.com/develop/develop-images/multistage-build/#name-your-build-stages), and my intuition, the `as` would be clearest to be capitalized to highlight it as a Docker keyword.
```suggestion
FROM rust AS builder
```